server/src/main/java/org/apache/iotdb/db/metadata/idtable/entry/DiskSchemaEntry.java

```diff
@@ -100,10 +103,31 @@ public static DiskSchemaEntry deserialize(InputStream inputStream) throws IOExce
     // read byte len
     res.entrySize = ReadWriteIOUtils.readInt(inputStream);
     res.entrySize += Integer.BYTES;
-
+    if (isTombstone(res.deviceID)) res.deviceID = TOMBSTONE;
     return res;
   }
 
+  public static void writeTombstone(RandomAccessFile randomAccessFile, long offset)
+      throws IOException {
+    randomAccessFile.seek(offset);
+    int strLength = randomAccessFile.readInt();
+    byte[] bytes = new byte[strLength];
+    randomAccessFile.write(bytes, 0, strLength);
+  }
+
+  public static String readString(RandomAccessFile randomAccessFile) throws IOException {
+    int strLength = randomAccessFile.readInt();
+    byte[] bytes = new byte[strLength];
+    randomAccessFile.read(bytes, 0, strLength);
+    return new String(bytes, 0, strLength);
+  }
```

Comment:
This class should only sense the byte data rather than the file object. Remove randomAccessFile from this class.

Reply:
fixed